libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -739,14 +669,44 @@ static IotMqttError_t _subscriptionCommon( IotMqttOperationType_t operation,
         *pOperationReference = pSubscriptionOperation;
     }
 
-    /* Schedule the subscription operation for network transmission. */
-    status = _IotMqtt_ScheduleOperation( pSubscriptionOperation,
-                                         _IotMqtt_ProcessSend,
-                                         0 );
+    if( operation == IOT_MQTT_SUBSCRIBE )
+    {
+        /* Calling SUBSCRIBE wrapper to send SUBSCRIBE packet on the network using MQTT LTS SUBSCRIBE API. */
+        status = _IotMqtt_managedSubscribe( mqttConnection,
+                                            pSubscriptionOperation,
+                                            pSubscriptionList,
+                                            subscriptionCount );
+    }
 
-    if( status != IOT_MQTT_SUCCESS )
+    if( operation == IOT_MQTT_UNSUBSCRIBE )
```

Comment:
else if?

Reply:
The operation will be either subscribe or unsubsribe and the inscribe operation type is checked after this condition, so else if is not required in this case.